nixos/modules/services/misc/adaptivemm.nix

```
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adaptivemm;
+in {
+  options = {
+    services.adaptivemm = {
+      enable = mkEnableOption (lib.mdDoc "Proactively tune kernel free memory configuration");
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.adaptivemm;
+        defaultText = literalExpression "pkgs.adaptivemm";
+        description = lib.mdDoc ''
+          Which adaptivemm package to use.
+        '';
+      };
+
+      verbosity = mkOption {
+        type = types.int;
```

There's some magic stuff in `lib` that let's us avoid the nasty asserts.

```suggestion
        type = types.int.between 0 5;
```